diff -Naur a/Makefile.linux_sdl b/Makefile.linux_sdl
--- a/Makefile.linux_sdl	2018-02-22 22:34:30.000000000 -0500
+++ b/Makefile.linux_sdl	2021-03-25 22:41:45.319866569 -0400
@@ -1,6 +1,6 @@
 # make -B -f Makefile.linux_sdl
 
-CC = gcc
+CC := gcc
 LD = $(CC)
 
 CFLAGS = -O3
@@ -21,7 +21,7 @@
 
 ## NOTE inc paths are from TinyCore SDL-dev
 DEFS = -DIS_LITTLE_ENDIAN -DALT_PATH_SEP
-INCS = -I$(GNUBOY) -I$(GNUBOY)/sys/sdl -I/usr/include/ -I/tmp/tcloop/SDL-dev/usr/local/
+INCS = -I$(GNUBOY) -I$(GNUBOY)/sys/sdl -I$(STAGING_DIR)/usr/include/
 LIBS =  -lSDLmain -lSDL
 
 # Use SDL_image - for use with SFont (rather than FREETYPE_TTF)
@@ -41,7 +41,7 @@
 # Simple disable sound support
 # not as cleanly implemented as in core gnuboy as it
 # still includes unused pcm/sound code... But it works :-)
-GNUBOY_DISABLE_SDL_SOUND = True
+#GNUBOY_DISABLE_SDL_SOUND = True
 ifdef GNUBOY_DISABLE_SDL_SOUND
 	OBJS += $(GNUBOY)/sys/dummy/nosound.o
 	DEFS += -DGNUBOY_DISABLE_SDL_SOUND
