diff --git a/gfx/drivers/sdl_gfx.c b/gfx/drivers/sdl_gfx.c
index 3e9f4a7342..882fcfef3c 100644
--- a/gfx/drivers/sdl_gfx.c
+++ b/gfx/drivers/sdl_gfx.c
@@ -376,12 +376,18 @@ static bool sdl_gfx_frame(void *data, const void *frame, unsigned width,
    if (!frame)
       return true;
 
-   title[0] = '\0';
+   //title[0] = '\0';
 
-   video_driver_get_window_title(title, sizeof(title));
+   //video_driver_get_window_title(title, sizeof(title));
 
-   if (SDL_MUSTLOCK(vid->screen))
-      SDL_LockSurface(vid->screen);
+   if (vid->menu.active) {
+      #ifdef HAVE_MENU
+         menu_driver_frame(menu_is_alive, video_info);
+      #endif
+      SDL_BlitSurface(vid->menu.frame, NULL, vid->screen, NULL);
+   } else {
+      if (SDL_MUSTLOCK(vid->screen))
+        SDL_LockSurface(vid->screen);
 
    video_frame_scale(
          &vid->scaler,
@@ -395,24 +401,17 @@ static bool sdl_gfx_frame(void *data, const void *frame, unsigned width,
          height,
          pitch);
 
-#ifdef HAVE_MENU
-   menu_driver_frame(menu_is_alive, video_info);
-
-   if (vid->menu.active)
-      SDL_BlitSurface(vid->menu.frame, NULL, vid->screen, NULL);
-#endif
-
-   if (msg)
-      sdl_render_msg(vid, vid->screen,
-            msg, vid->screen->w, vid->screen->h, vid->screen->format,
+       if (msg)
+         sdl_render_msg(vid, vid->screen, msg, vid->screen->w, vid->screen->h, vid->screen->format,
             video_info->font_msg_pos_x,
             video_info->font_msg_pos_y);
 
    if (SDL_MUSTLOCK(vid->screen))
       SDL_UnlockSurface(vid->screen);
+   }
 
-   if (title[0])
-      SDL_WM_SetCaption(title, NULL);
+   //if (title[0])
+   //   SDL_WM_SetCaption(title, NULL);
 
    SDL_Flip(vid->screen);
 
